provisioning/WFGP-292_doc_galleon-pack.adoc

```diff
@@ -0,0 +1,164 @@
+
+However, over time, this approach has shown some cracks:
+
+* Some new subsystems are provided by other configurations (e.g. standalone-microprofile.xml). This was worked around by adding additional extensions when the model referenced is generated (in WildFly's `docs` module) but is out of sync.
+* WildFly is now composed of 2 Galleon packs (wildfly-ee-galleon-pack & wildfly-galleon-pack). Each Galleon pack provides different extensions and resources but this is not clear from the management model (e.g. which Galleon pack provides the `microprofile-config-smallrye` subsystem?)
+* Additional capabilities for WildFly are provided by other feature packs (mvc-krazo, grpc, wildfly-preview, etc.) that do not provide their model references and makes it more difficult for the users to understand which resources, attributes and operations are available to configure their installations. They have to rely on jboss-cli to get the information and while it is available, it does not make it very user-friendly.
+* More generally, the tool that is used to generate this model reference (wildscribe) is lagging behind new WildFly management features (stability level was added recently while it's been in WildFly for some time)
+
+We can improve the user experience for WildFly users by consolidating our tooling and providing consistent documentation for the WildFly ecosystem.
+
+Each feature pack should generate user documentation that describes the management capabilities offered by the feature pack in a user-friendly readable output.
+This user documentation should be made available in https://docs.wildfly.org/ or the web site that documents other feature packs.
+
+This documentation should be generated when the feature pack is built and made available as a companion to the feature pack. This should not required additional work from the feature pack developer other than providing correct information about their feature packs.
+
+=== User Stories
+
+* As a WildFly user, I want to have documentation of the capabilities and resources that are provided by any feature packs from WildFly community.
+* As a feature pack developer, I want to provide documentation for my feature pack that is automatically generated whenever I build my feature pack.
+
+== Issue Metadata
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFGP-292[WFGP-292 - Generate documentation for Feature Pack]
+
+=== Affected Projects or Components
+
+* WildFly Galleon Plugins - https://github.com/wildfly/galleon-plugins
+** Generate documentation when the `build-feature-pack` goal is invoked. It generates a companion doc archive next to the feature pack itself
+* Galleon - https://github.com/wildfly/galleon
+** Might be impacted if we are missing any metadata from the feature pack to generate its documentation
+* Wildscribe - https://github.com/wildfly/wildscribe
+** This RFE would be a full replacement of Wildscribe that would be archived.
```

Comment:
I think archiving Wildscribe should be out of scope. It is used downstream and we shouldn't mix downstream-impacting changes into this, as doing so just adds unnecessary friction.

I suggest replacing "that would be archived" with a sub-bullet saying "The https://github.com/wildfly/wildscribe/blob/main/README.md will be updated to state that the project is in maintenance mode and is no longer used by the WildFly application server itself."

Reply:
I agree, this statement was too aggressive. I've used your suggestion.

Regarding downstream uses, this RFE could also be interesting to apply to downstream feature packs that are lacking this kind of documentation at the moment too. But that's out of scope of what we want to achieve in that proposal